src/background/bgPopupHandler.ts

```diff
@@ -80,3 +83,28 @@ export const openPopup = async (
 export const removePopup = (id: string) => delete popups[id];
 
 export const getPopup = (id: string): IPopupConfigNoActions => popups[id];
+
+export const setSessionExpiration = async (expires: number) => {
+  await storageSession.set({ sessionExpires: expires });
+};
+
+export const getSession = async (): Promise<IExportedKey | null> => {
+  try {
+    const { sessionExpires } = await storageSession.get('sessionExpires');
+    if (sessionExpires < Date.now() || !sessionExpires) {
```

Comment:
I suggest to always start from conditions that are easier to calculate:

```suggestion
    if (!sessionExpires || sessionExpires < Date.now()) {
```